File: PHP/CompraFacade.php
<?php

require "./Estoque.php";
require "./Entrega.php";
require "./Pagamento.php";

class CompraFacade{
    public static function finalizarCompra($idProduto,$endereco,$cep,$transportadora,$meioDePagamento){
        $valorDoProduto = Estoque::retornarPrecoDoProdutoPeloID($idProduto);
        $entrega = new Entrega();
        $entrega->endereco = $endereco;
        $entrega->cep = $cep;
        $entrega->transportadora = $transportadora;
        $valorDoFrete = $entrega->calcularFrete();
        $valorTotal = $valorDoProduto + $valorDoFrete;
        if($meioDePagamento == 1){
            Pagamento::pagarComCartao($valorTotal);
        }else{
            Pagamento::pagarComBoleto($valorTotal);
        }
    }
}

File: Java/CompraFacade.java
package teste;

public class CompraFacade {
	public static void FinalizarCompra(int idProduto, String endereco, String cep, String transportadora, int meioDePagamento) {
		float precoDoProduto = Estoque.retornarPrecoDeUmProdutoPeloID(idProduto);
		Entrega correios = new Entrega();
		correios.endereco = endereco;
		correios.cep = cep;
		correios.transportadora = transportadora;
		float precoDoFrete = correios.calcularFrete();
		float total = precoDoProduto + precoDoFrete;
		if(meioDePagamento == 1) {
			Pagamento.pagarPeloCartao(total);
		}else {
			Pagamento.pagarPeloBoleto(total);
		}
	}	
}
